pkg/collector/corechecks/cluster/orchestrator/manifest_buffer.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package orchestrator
+
+import (
+	"time"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/collectors"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	defaultFlushManifestTime = 20 * time.Second
+	BufferManifest           = true
+)
+
+// ManifestBuffer is a buffer of manifest sent from all collectors
+// It has a slice bufferedManifests used to buffer manifest and stop channel
+// ManifestBuffer is started as a dedicated thread each time CollectorBundle runs a check
+// and gets stopped after the check is done.
+type ManifestBuffer struct {
+	Cfg               *collectors.ManifestBufferConfig
+	bufferedManifests []interface{}
+	stopCh            chan struct{}
+}
+
+// NewManifestBuffer returns a new ManifestBuffer
+func NewManifestBuffer(chk *OrchestratorCheck) *ManifestBuffer {
+	manifestBuffer := &ManifestBuffer{
+		Cfg: &collectors.ManifestBufferConfig{
+			ClusterID:                chk.clusterID,
+			KubeClusterName:          chk.orchestratorConfig.KubeClusterName,
+			MsgGroupRef:              chk.groupID,
+			MaxPerMessage:            chk.orchestratorConfig.MaxPerMessage,
+			MaxWeightPerMessageBytes: chk.orchestratorConfig.MaxWeightPerMessageBytes,
+			ManifestChan:             make(chan *model.Manifest),
+			BufferedManifestEnabled:  BufferManifest,
+			MaxBufferedManifests:     2 * chk.orchestratorConfig.MaxPerMessage,
+		},
+		stopCh: make(chan struct{}),
+	}
+	manifestBuffer.bufferedManifests = make([]interface{}, 0, manifestBuffer.Cfg.MaxBufferedManifests)
+
+	return manifestBuffer
+}
+
+// flushManifest flushes manifests by chunking them first then sending them to the sender
+func (cb *ManifestBuffer) flushManifest(sender aggregator.Sender) {
+	manifests := cb.bufferedManifests
+	cb.bufferedManifests = []interface{}{}
```

Comment:
you are resetting it after flushing, right? Why not reset it to the given size as above? Alternatively, I think cb.bufferedManifests = `cb.bufferedManifests[:0]` should work to re-use the backing array